pkg/vfs/vfs.go

```diff
@@ -94,6 +94,9 @@ type Indexer interface {
 
 	// CreateFileDoc creates and add in the index a new file document.
 	CreateFileDoc(doc *FileDoc) error
+	// CreateNamedFileDoc creates and add in the index a new file document with
+	// its id already set.
+	CreateNamedFileDoc(doc *FileDoc) error
```

Comment:
Let's leave it like that, it's ok for me.

I was wrong about the collisions, CouchDB should provide IDs with small collision probabilities.